cmake/create_test.cmake

```diff
@@ -1,22 +1,32 @@
-set(gko_test_single_args "MPI_SIZE")
+set(gko_test_resource_args "RESOURCE_LOCAL_CORES;RESOURCE_PERCENTAGE;RESOURCE_TYPE")
+set(gko_test_single_args "MPI_SIZE;${gko_test_resource_args}")
 set(gko_test_multi_args "DISABLE_EXECUTORS;ADDITIONAL_LIBRARIES;ADDITIONAL_INCLUDES")
+set(gko_test_option_args "NO_RESOURCES")
 
 ## Replaces / by _ to create valid target names from relative paths
 function(ginkgo_build_test_name test_name target_name)
     file(RELATIVE_PATH REL_BINARY_DIR
-        ${PROJECT_BINARY_DIR} ${CMAKE_CURRENT_BINARY_DIR})
+            ${PROJECT_BINARY_DIR} ${CMAKE_CURRENT_BINARY_DIR})
     string(REPLACE "/" "_" TEST_TARGET_NAME "${REL_BINARY_DIR}/${test_name}")
     set(${target_name} ${TEST_TARGET_NAME} PARENT_SCOPE)
-endfunction(ginkgo_build_test_name)
+endfunction()
+
+function(ginkgo_create_gtest_main)
```

Comment:
we don't seem to be using the targets

Reply:
Yes, due to the changes of the gtest-main we no longer compile the main separately. Each test now compiles its own main, since the main uses different compile definitions depending on the test type. Still, I hope that by using ccache that should not add too much of an overhead.

Reply:
you could take the `TARGET_FILES` object library approach we use in our executors to avoid duplication

Reply:
would that help? I would need to create 10 (5 per executor, 2 per mpi) object libraries, and then I could just use normal libraries for that.

Reply:
good point, then no 😆